src/Tools/dotnet-sos/Program.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using McMaster.Extensions.CommandLineUtils;
+using SOS;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.SOS
+{
+    [Command(Name = "dotnet-sos", Description = "Install and configure SOS")]
+    internal class Program
+    {
+        [Option("--install", Description = "Install and configure SOS.")]
+        public bool InstallSOS { get; set; }
+
+        [Option("--uninstall", Description = "Uninstall SOS.")]
+        public bool UninstallSOS { get; set; }
+
+        [Option("--source", Description = "SOS binaries source path.")]
+        public string SOSSourcePath { get; set; }
+
+        public int OnExecute(IConsole console, CommandLineApplication app)
+        {
+            if (InstallSOS || UninstallSOS)
+            {
+                var sosInstaller = new InstallHelper((message) => console.WriteLine(message));
+                if (SOSSourcePath != null)
+                {
+                    sosInstaller.SOSSourcePath = SOSSourcePath;
+                }
+                try
+                {
+                    if (UninstallSOS)
+                    {
+                        sosInstaller.Uninstall();
+                    }
+                    else 
+                    {
+                        sosInstaller.Install();
+                    }
+                }
+                catch (Exception ex) when (ex is ArgumentException || ex is PlatformNotSupportedException || ex is InvalidOperationException)
```

Comment:
Sorry I was a little ambiguous in what I said before. What I intended to suggest was to define a new exception type, for example CommandLineErrorException, and then this catch becomes:

    catch(CommandLineErrorException ex)
    {
        console.Error.WriteLine(ex.Message);
    }

This makes it obvious that any strings that go into that exception type are part of the user-facing error handling text and all other exceptions are bugs/error conditions we didn't properly anticipate. My worry for how the code is now is that it is too hard to ensure that the framework won't throw its own exception of type ArgumentException with a useless error message such as "length".

Reply:
This exception would have to be something SOS installer specific like SOSInstallerException because of the layering I'm envisioning that the install helper doesn't know anything about the command line parsing, etc.

Reply:
Sounds good to me.